src/workspacer.Native/Native/WindowsManager.cs

```diff
@@ -298,16 +304,26 @@ private void WindowMove(IntPtr handle)
                 var window = _windows[handle];
                 if (_mouseMoveWindow == window)
                 {
-                    WindowUpdated?.Invoke(window, WindowUpdateType.Move);
+                    UpdateWindowTrigger?.Invoke(window, WindowUpdateType.Move);
                 }
             }
         }
 
-        private void HandleWindowFocused(WindowsWindow window)
+        private void HandleWindowFocused(IWindow window)
         {
             WindowFocused?.Invoke(window);
         }
 
+        private void HandleWindowUpdated(IWindow window)
+        {
+            WindowUpdated?.Invoke(window);
+        }
+
+        private void HandleWindowClosed(IWindow window)
+        {
+            WindowClosed?.Invoke(window);
+        }
+
         private void HandleWindowMoveStart(WindowsWindow window)
```

Comment:
Any reason `window` here needs to be a `WindowsWindow` instead of just a `IWindow`?

This particular method stands out :smile:

Reply:
The issue is that IsMouseMoving is only a Getter in the IWindow class and it is set a few times throughout this class

Reply:
Right. So it's deliberate and not an oversight. Sounds good to me :smile: